ecommerce/pricing/lib/pricing/helpers/happy_hours_for_product.rb

```diff
@@ -51,11 +50,21 @@ def build_happy_hour(start_hour, end_hour)
           first_segment = Range.new(0, end_hour - 1).to_a
           second_segment = Range.new(start_hour, 23).to_a
 
+          validate_hours_list(first_segment)
+          validate_hours_list(second_segment)
+
           first_segment.union(second_segment)
         else
-          Range.new(start_hour, end_hour - 1).to_a
+          Range.new(start_hour, end_hour - 1)
         end
       end
+
+      def validate_hours_list(sorted_hours)
```

Comment:
Added because otherwise mutant was complaining that he could change 23 to 24 on line 51 and nothing was failing. I don't see other way of triggering the error in tests.